lib/internal/util.js

```diff
@@ -570,6 +570,13 @@ function isArrayBufferDetached(value) {
   return false;
 }
 
+function getUtf8Length(str) {
+  let length = 0;
+  for (let i = 0; i < str.length; i++)
+    length += str.charCodeAt(i) > 127 ? 2 : 1;
```

Comment:
How is this correct? Is there a restriction on the code points in the input `str`?

Reply:
I don’t we need this since I’m trying a new approach which does not require utf8 calculation but assumes it.